core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -470,10 +473,15 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
 
         createContainerCmdModifiers.forEach(hook -> hook.accept(createCommand));
 
-        Map<String, String> labels = createCommand.getLabels();
-        labels = new HashMap<>(labels != null ? labels : Collections.emptyMap());
-        labels.putAll(DockerClientFactory.DEFAULT_LABELS);
-        createCommand.withLabels(labels);
+        Map<String, String> createCommandLabels = createCommand.getLabels();
+        createCommandLabels = new HashMap<>(createCommandLabels != null ? createCommandLabels : Collections.emptyMap());
+
+        Map<String, String> combinedLabels = new HashMap<>();
+        combinedLabels.putAll(labels);
+        combinedLabels.putAll(DockerClientFactory.DEFAULT_LABELS);
+        combinedLabels.putAll(createCommandLabels);
```

Comment:
`DEFAULT_LABELS` must always override :)

Reply:
Also, I would change to:
```java
if (createCommand.getLabels() != null) {
    combinedLabels.putAll(createCommand.getLabels());
}
```
to avoid a variable

Reply:
So the order should be `user defined labels` < `createCommandLabels` < `DEFAULT_LABELS`?

Reply:
Yes :) Sorry if I confused you about it before!

Reply:
No worries ;)